pyaerocom/colocation/colocated_data.py

```diff
@@ -856,38 +856,31 @@ def check_set_countries(self, inplace=True, assign_to_dim=None):
             name of dimension to which the country coordinate is assigned.
             Default is None, in which case station_name is used.
 
-        Raises
-        ------
-        DataDimensionError
-            If data is 4D (i.e. if latitude and longitude are othorgonal
-            dimensions)
-
         Returns
         -------
         ColocatedData
             data object with countries assigned
 
         """
-        if self.has_latlon_dims:
-            raise DataDimensionError(
-                "Countries cannot be assigned to 4D"
-                "ColocatedData with othorgonal lat / lon "
-                "dimensions. Please consider stacking "
-                "the latitude and longitude dimensions-"
-            )
         if assign_to_dim is None:
             assign_to_dim = "station_name"
 
-        if assign_to_dim not in self.dims:
-            raise DataDimensionError("No such dimension", assign_to_dim)
+        # stack lat and lon dimensions into station_name for 4D data
+        if all(_ in self.dims for _ in ("latitude", "longitude")):
+            coldata = self if inplace else self.copy()
+            logger.info("Stacking lat and lon dimensions into station_name")
+            coldata = coldata.stack(station_name=("latitude", "longitude"))
```

Comment:
The inplace code was very clean (one line), but is now a mess, merged all over this function.

Do we need this function inplace/non-inplace functionality? Do we have tests for it (I guess it uses mostly the default==inplace)? Maybe we should remove?

Reply:
Doesn't look like it's used anywhere else. We could just get rid of it, but that changes the interface. Happy to get rid of it if you want.

Reply:
Feel free to remove it.